lib/routes/session.js

```diff
@@ -142,6 +142,11 @@ module.exports = function (log, db, Password, config, signinUtils) {
               if (result) {
                 verificationMethod = 'totp-2fa'
               }
+
+              // Error if requesting TOTP verification with TOTP not setup
+              if (! result && verificationMethod === 'totp-2fa') {
```

Comment:
Probably six of one and half a dozen of the other, but I'm looking at this wondering if the flow would be clearer if it was an `else if (verificationMethod === 'totp-2fa')` instead of a fresh `if (! result && ...)`. (and above too)